### ISSUE TITLE
Node Count increases up to 18, then drops to 8 at once, then 18 and repeat

### ISSUE THREAD

**Issue by PastaPastaPasta** [dashpay/dash#2208]
/re: title

[Playback at 4x speed ](https://youtu.be/50YziGNe110)

The node count starts at 18, then drops to 8, then increases back up to 18 and this loops about a million times. I'm pretty sure 8 is the max node count, so maybe the add node function doesn't check if it can add it without going over the limit, and then when the limit is checked some are removed. But I'm really not sure what's happening...

**Comment by jhodges10** [comment]
Looks this occurs when each new block is mined. I would be inclined to think this has something to do with masternode payee selection as well as rotating through connected masternodes to keep the quorums well mixed and distributed. @UdjinM6 you probably know exactly what's going on here.

**Comment by UdjinM6** [comment]
TL;DR: it's a feature, not a bug ;)

That's how "masternode recovery" works. Your node requests from top10 masternodes `mnb`s for some other masternodes it has in `NEW_START_REQUIRED` state and if majority of top10 masternodes relay you `mnb` with a very recent `mnp` (i.e. corresponding masternode is in `ENABLED` state for them), then your node also updates local stats and switch such masternode into `ENABLED` mode. This is needed because initial sync is not deterministic and some nodes you connect to on start might relay you incorrect info.

**Comment by PastaPastaPasta** [comment]
Okay, sounds good. It just seemed odd
